tests/app/test_prompt_api.py

```diff
@@ -62,8 +62,37 @@ def test_app_prompt_validation_errors(mock_write_json_to_file):
         "context": [{"retrieved_from": "test_data.pdf"}],
     }
     response = client.post("/v1/prompt", json=test_payload)
-    assert response.status_code == 500
-    assert response.json()["detail"] == "doc is a mandatory field"
+    assert response.status_code == 400
+    assert response.json()["detail"] == "doc, vector_db are mandatory fields."
```

Comment:
1. Why is there no error for the missing `prompt_time`?
2. The locations of `doc` and `vector_db` are not clear in the message. Are they missing at the root level or inside some entry in the context?

Reply:
Updated accordingly.
Now error message will be like:
```
"4 validation errors for RetrievalData\n"
"context -> 0 -> vector_db\n"
"  field required (type=value_error.missing)\n"
"context -> 1 -> doc\n"
"  field required (type=value_error.missing)\n"
"prompt_time\n"
"  field required (type=value_error.missing)\n"
"user\n"
"  field required (type=value_error.missing)"
```